src/main/webapp/resources/js/components/samples/components/SampleFiles.tsx

```diff
@@ -55,27 +62,31 @@ export default function SampleFiles() {
 
   /*
    Get the sample files and set them in the redux store on component load
-   and to refetch if the sample identifier or project change
+   and to refetch if any of the dependencies in the dependency array change
    */
   React.useEffect(() => {
-    dispatch(
-      fetchFilesForSample({
-        sampleId: sample.identifier,
-        projectId,
-      })
-    );
-  }, [sample.identifier, projectId, dispatch]);
+    if (Object.keys(files).length > 0) {
+      dispatch(
+        fetchFilesForSample({
+          sampleFiles: files,
+        })
+      );
+    }
+  }, [sample.identifier, projectId, dispatch, loading, files]);
 
   /*
   Call function to upload files to server once files are
   selected through the open file dialog or if they are
   dragged and dropped
    */
-  React.useEffect(() => {
-    if (filesToUpload.length) {
-      uploadFiles();
-    }
-  }, [filesToUpload]);
+  React.useEffect(
+    () => {
+      if (filesToUpload.length) {
+        uploadFiles();
+      }
+    }, // eslint-disable-next-line react-hooks/exhaustive-deps
+    [filesToUpload]
```

Comment:
**CHANGE THIS**: Don't need the eslint-disabled-nex-line.  Your function wont change so you can add it as a Don't.  You will have to move the declaration above the useEffect hook because it is an arrow function and does not get hoisted dependency.  You will also need to wrap it in a useCallback.

Reply:
Thanks for walking me through this. Updated in 152ba864e8160e1aec8276f84e433db80e1d2f1b

Reply:
Better, thanks ;)